scripts/f5-cookie-decode.nse

```diff
@@ -0,0 +1,70 @@
+-- See here: https://support.f5.com/csp/article/K6917
+description = [[
+Decodes any unencrypted F5 BIG-IP cookies in the HTTP response.
+]]
+
+local http = require "http"
+local shortport = require "shortport"
+local stdnse = require "stdnse"
+local string = require "string"
+local utf8 = require "utf8"
+
+---
+-- @usage
+-- nmap -p <port> --script f5-cookie-decode <target>
+--
+-- @output
+-- PORT    STATE SERVICE
+-- 80/tcp  open  http
+-- | f5-cookie-decode:
+-- |_  BIGipServer<pool_name>:10.1.1.100:8080
+--
+-- @args f5-cookie-decode.path The URL path to request. The default path is "/".
+
+author = "Seth Jackson"
+license = "Same as Nmap--See https://nmap.org/book/man-legal.html"
+categories = { "discovery", "safe" }
+
+portrule = shortport.http
+
+action = function(host, port)
+  local path = stdnse.get_script_args(SCRIPT_NAME..".path") or "/"
+
+  local response = http.get(host, port, path)
+
+  if not response then
+    return
+  end
+
+  if not response.cookies then
+    return
+  end
+
+  local decoded = stdnse.output_table()
+
+  for _, cookie in pairs(response.cookies) do
+    if cookie.name:find("BIGipServer") then
+      local split = stdnse.strsplit("%.", cookie.value)
+
+      local host = tonumber(split[1])
+      local port = split[2]
+
+      if host then
+        local packed = string.pack("<I4", host)
+        local values = {}
+
+        for i = 1, #packed do
+          local c = packed:sub(i, i)
+          table.insert(values, utf8.codepoint(c))
+        end
+
+        local ip_address = stdnse.strjoin(".", values)
+        port = tonumber(stdnse.tohex(string.pack("<H", port)), 16)
+
+        table.insert(decoded, string.format("%s:%s:%s", cookie.name, ip_address, port))
```

Comment:
If you are ambitious, it would be nice if the XML output was truly structured. For cookie `BIGipServeragdncdcqd05_pool=1193829386.24866.0000` this could be something like:
```xml
<cookie>
  <pool>agdncdcqd05_pool</pool>
  <address>
    <addr>10.100.40.71</addr>
    <type>ipv4</type>
  </address>
  <port>8801</port>
</cookie>
```
(Adding the IPv4 designation, which might look overbearing right now, also gives you the opportunity to add decoding for IPv6 cookies in the future without breaking backward compatibility of the output .)

You can still keep your current compact format for the regular textual output by maintaining two tables, one for the XML output and one for the textual, and then return the findings as follows:
```lua
  if #output > 0 then
    return output, stdnse.format_output(true, text_output)
  end
```
If you do decide to add the XML output then do not forget to add the corresponding XML output example to the documentation section above.